client/src/Pages/PlayerWaitingScreen.jsx

```diff
@@ -1,11 +1,129 @@
 import React from 'react';
-
+import {Button, Row, Col, Modal} from 'antd';
+import "../Styles/Waiting.scss";
+import RulesPage from './RulesPage';
+import CharcterPage from './CharacterPage';
 export default class PlayerWaitingScreen extends React.Component{
+    
+    state = {
+        rulesVisible: false,
+        charactersVisible: false,
+    };
```

Comment:
Don't you set the initial state in the constructor? That's what's shown [here](https://reactjs.org/docs/state-and-lifecycle.html#adding-local-state-to-a-class).